conformance/experimental_conformance_test.go

```diff
@@ -0,0 +1,151 @@
+	exemptFeatures = suite.ParseSupportedFeatures(*flags.ExemptFeatures)
+	skipTests = suite.ParseSkipTests(*flags.SkipTests)
+	namespaceLabels = suite.ParseNamespaceLabels(*flags.NamespaceLabels)
+
+	// experimental conformance flags
+	conformanceProfiles := suite.ParseConformanceProfiles(*flags.ConformanceProfiles)
+
+	if conformanceProfiles.Len() > 0 {
+		// if some conformance profiles have been set, run the experimental conformance suite...
+		implementation, err = suite.ParseImplementation(
+			*flags.ImplementationOrganization,
+			*flags.ImplementationProject,
+			*flags.ImplementationUrl,
+			*flags.ImplementationVersion,
+			*flags.ImplementationContact,
+		)
+		if err != nil {
+			t.Fatalf("Error parsing implementation's details: %v", err)
+		}
+		testExperimentalConformance(t)
+	} else {
+		// ...otherwise run the standard conformance suite.
+		t.Run("standard conformance tests", TestConformance)
+	}
+}
+
+func testExperimentalConformance(t *testing.T) {
+	t.Logf("Running experimental conformance tests with %s GatewayClass\n cleanup: %t\n debug: %t\n enable all features: %t \n supported features: [%v]\n exempt features: [%v]",
+		*flags.GatewayClassName, *flags.CleanupBaseResources, *flags.ShowDebug, *flags.EnableAllSupportedFeatures, *flags.SupportedFeatures, *flags.ExemptFeatures)
+
+	cSuite, err := suite.NewExperimentalConformanceTestSuite(
+		suite.ExperimentalConformanceOptions{
+			Options: suite.Options{
+				Client:     k8sClient,
+				RESTClient: clientset.CoreV1().RESTClient().(*rest.RESTClient),
+				RestConfig: cfg,
+				// This clientset is needed in addition to the client only because
+				// controller-runtime client doesn't support non CRUD sub-resources yet (https://github.com/kubernetes-sigs/controller-runtime/issues/452).
+				Clientset:                  clientset,
+				GatewayClassName:           *flags.GatewayClassName,
+				Debug:                      *flags.ShowDebug,
+				CleanupBaseResources:       *flags.CleanupBaseResources,
+				SupportedFeatures:          supportedFeatures,
+				ExemptFeatures:             exemptFeatures,
+				EnableAllSupportedFeatures: *flags.EnableAllSupportedFeatures,
+				NamespaceLabels:            namespaceLabels,
+				SkipTests:                  skipTests,
+			},
+			Implementation:      *implementation,
+			ConformanceProfiles: conformanceProfiles,
+		})
+	if err != nil {
+		t.Fatalf("error creating experimental conformance test suite: %v", err)
+	}
+
+	cSuite.Setup(t)
+	cSuite.Run(t, tests.ConformanceTests)
+	report, err := cSuite.Report()
+	if err != nil {
+		t.Fatalf("error generating conformance profile report: %v", err)
+	}
+	writeReport(t.Log, *report, *flags.ReportOutput)
+}
+
+func writeReport(log func(...any), report confv1a1.ConformanceReport, output string) error {
+	rawReport, err := yaml.Marshal(report)
+	if err != nil {
+		return err
+	}
+
+	if output != "" {
+		if err = os.WriteFile(output, rawReport, 0666); err != nil {
```

Comment:
The file doesn't need to be _writable_ by group/world, we can probably just give write permissions to the file owner (current user for the test program):

```suggestion
		if err = os.WriteFile(output, rawReport, 0644); err != nil {
```